src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.Expander.cs

```diff
@@ -271,6 +294,60 @@ public override SyntaxNode VisitArgument(ArgumentSyntax node)
                 return newArgument;
             }
 
+            private static bool CanMakeNameExplicitInTuple(TupleExpressionSyntax tuple, string name)
+            {
+                if (name == null || SyntaxFacts.IsReservedTupleElementName(name, out _))
+                {
+                    return false;
+                }
+
+                bool found = false;
+                foreach (var argument in tuple.Arguments)
+                {
+                    string elementName = null;
+                    if (argument.NameColon != null)
+                    {
+                        elementName = argument.NameColon.Name.Identifier.ValueText;
+                    }
+                    else
+                    {
+                        elementName = argument.Expression.TryGetInferredMemberName().ValueText;
+                    }
+
+                    if (elementName?.Equals(name, StringComparison.Ordinal) == true)
+                    {
+                        if (found)
+                        {
+                            // No duplicate names allowed
+                            return false;
+                        }
+                        found = true;
+                    }
+
+                }
+
+                return true;
+            }
+
+            public override SyntaxNode VisitAnonymousObjectMemberDeclarator(AnonymousObjectMemberDeclaratorSyntax node)
+            {
+                var inferredName = node.Expression.TryGetInferredMemberName().ValueText;
```

Comment:
not needed until you verify that .NameEquals is null.  move inside lower if block.